immutable-history/test/VersionRepositoryTest.php

```diff
@@ -76,9 +86,550 @@ public function testPublishWithoutResourceIdOverrideAndWithFindablePreviousPubli
             return true;
         };
 
-        $entityManager->shouldReceive('persist')->withArgs($checkNewVersionEntity);
-        $entityManager->shouldReceive('flush')->withArgs($checkNewVersionEntity);
+        $entityManager->expects('persist')->withArgs($checkNewVersionEntity);
```

Comment:
this doesn't check that these methods are called in order. you should add ->ordered() to the end of each one.